src/worker/controller/actions/tasks/VerifyAndStoreResultAction.js

```diff
@@ -133,5 +103,43 @@ class VerifyAndStoreResultAction {
       };
     }
   }
+  async _verifyResult(resultObj, type, contractAddress) {
+    // TODO: remove this default!!!!
+    let isVerified = true;
+    let result = null;
+    let error = null;
+
+    if (this._controller.hasEthereum()) {
+      isVerified = false;
+      let result;
+      if (resultObj.status === constants.TASK_STATUS.FAILED) {
+        result = FailedResult.buildFailedResult(resultObj);
+      }
+      else {
+        if (type === constants.CORE_REQUESTS.DeploySecretContract) {
+          result = DeployResult.buildDeployResult(resultObj);
+        }
+        else {
```

Comment:
I would've put a check here that it's Compute, and else would give an error, because someone can misuse this.

Reply:
Thanks, this was treated at [this commit](https://github.com/enigmampc/enigma-p2p/commit/38d239d7161107958b11a25291d8598fd18ef601).
Thanks!